agent/token_store_test.go

```diff
@@ -0,0 +1,97 @@
+package agent
+
+import (
+	"testing"
+)
+
+func TestTokenStore_AgentToken(t *testing.T) {
+	t.Parallel()
+	tokens := new(TokenStore)
+
+	if got, want := tokens.GetTokenForAgent(), ""; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	// Precedence is agent token, user token, "".
+
+	tokens.UpdateUserToken("user")
+	if got, want := tokens.GetTokenForAgent(), "user"; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	tokens.UpdateAgentToken("agent")
+	if got, want := tokens.GetTokenForAgent(), "agent"; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	tokens.UpdateUserToken("")
```

Comment:
This does not cover when you clear the agent token first. I think you could write this as a table driven test

```go
func TestTokenStore_AgentToken(t *testing.T) {
    tests := []struct {
        user, agent, token string
    }{
        {"", "", ""},
        {"user", "", "user"},
        {"user", "agent", "agent"},
        {"", "agent", "agent"},
        {"user", "agent", "agent"},
        {"user", "", "user"},
        {"", "", ""},
    }
    tokens := new(TokenStore)
    for _, tt := range tests {
        tokens.UpdateUserToken(tt.user)
        tokens.UpdateAgentToken(tt.agent)
        if got, want := tokens.GetTokenForAgent(), tt.token; got != want {
            t.Fatalf("got token %q want %q", got, want)
        }
}
````